lang/cs/Org.Apache.REEF.Network/Group/Task/Impl/OperatorTopology.cs

```diff
@@ -392,7 +393,7 @@ private void WaitForTaskRegistration(IList<string> identifiers)
 
                 var leftOver = string.Join(",", identifiers.Where(e => !foundList.Contains(e)));
                 Logger.Log(Level.Error, "For node {0}, cannot find registered parent/children: {1}.", _selfId, leftOver);
-                Exceptions.Throw(new SystemException("Failed to initialize operator topology for node: " + _selfId), Logger);
+                Exceptions.Throw(new ReefRuntimeException("Failed to initialize operator topology for node: " + _selfId), Logger);
```

Comment:
You need to find all the places where this exception is caught and match the exception type. For example, this method is called in GroupCommClient constructor.  If you change it to ReefRuntimeException here, at GroupCommClient , you should also change it to the same type. Otherwise too general exceptions are caught. Why not use REEFSystemException at both places?